userspace/libsinsp/container_engine/docker_common.cpp

```diff
@@ -468,22 +469,25 @@ bool docker::resolve(container_cache_interface *cache, sinsp_threadinfo *tinfo,
 
 void docker::parse_docker_async(const string& container_id, container_cache_interface *cache)
 {
-	auto cb = [cache](const std::string& container_id, const sinsp_container_info& res)
-        {
+	m_cache = cache;
+
+	auto cb = [cache](const docker_async_instruction& instruction, const sinsp_container_info& res)
+	{
 		g_logger.format(sinsp_logger::SEV_DEBUG,
 				"docker_async (%s): Source callback result=%d",
-				container_id.c_str(),
+				instruction.container_id.c_str(),
 				res.m_lookup_state);
 
 		cache->notify_new_container(res);
 	};
 
         sinsp_container_info result;
 
-	if (m_docker_info_source->lookup(container_id, result, cb))
+	docker_async_instruction instruction(container_id, false /*don't request size*/);
```

Comment:
An enum instead of the bool + comment?

Reply:
I started to do this and the enum is basically yes/no; I think I'll leave this as an bool